plugin-niassa/src/main/java/ca/on/oicr/gsi/shesmu/niassa/MigrationAction.java

```diff
@@ -0,0 +1,287 @@
+
+public final class MigrationAction extends Action {
+
+  private static final LatencyHistogram matchTime =
+      new LatencyHistogram(
+          "shesmu_niassa_migrate_match_time",
+          "The time to match an action against the workflow runs in the database.",
+          "workflow");
+
+  static final Comparator<LimsKey> LIMS_ID_COMPARATOR =
+      Comparator.comparing(LimsKey::getProvider).thenComparing(LimsKey::getId);
+  static final Comparator<LimsKey> LIMS_KEY_COMPARATOR =
+      LIMS_ID_COMPARATOR.thenComparing(LimsKey::getVersion);
+
+  private final Map<String, String> annotations;
+  private boolean cacheCollision;
+  private List<String> errors = Collections.emptyList();
+  private final FileMatchingPolicy fileMatchingPolicy;
+  Properties ini = new Properties();
+  private InputLimsCollection limsKeysCollection;
+  private long majorOliveVersion;
+  private List<WorkflowRunMatch> matches = Collections.emptyList();
+  private final long[] previousAccessions;
+
+  @ActionParameter(required = false)
+  public long priority;
+
+  private boolean priorityBoost;
+  private final boolean relaunchFailedOnUpgrade;
+  final SubmitWorkflowRequest request = new SubmitWorkflowRequest();
+  private final Definer<NiassaServer> server;
+  private final Set<String> services;
+  private boolean stale;
+  private RunState state = new RunStateAttemptSubmit();
+  private final long workflowAccession;
+  private final String workflowName;
+
+  public MigrationAction(
+      Definer<NiassaServer> server,
+      String workflowName,
+      long workflowAccession,
+      long[] previousAccessions,
+      FileMatchingPolicy fileMatchingPolicy,
+      List<String> services,
+      Map<String, String> annotations,
+      boolean relaunchFailedOnUpgrade) {
+    super("niassa-migration");
+    this.server = server;
+    this.workflowName = workflowName;
+    this.workflowAccession = workflowAccession;
+    this.previousAccessions = previousAccessions;
+    this.fileMatchingPolicy = fileMatchingPolicy;
+    this.services = new TreeSet<>(services);
+    this.annotations = new TreeMap<>(annotations);
+    // This sort will group workflows together by SWID so that as many as possible will performed
+    // sequentially before analysis cache expires and needs reloading.
+
+    priority = (workflowAccession % 10);
+    this.relaunchFailedOnUpgrade = relaunchFailedOnUpgrade;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    MigrationAction that = (MigrationAction) o;
+
+    return workflowAccession == that.workflowAccession
+        && Objects.equals(annotations, that.annotations)
+        && Objects.equals(ini, that.ini)
+        && Objects.equals(limsKeysCollection, that.limsKeysCollection);
+  }
+
+  @Override
+  public void generateUUID(Consumer<byte[]> digest) {
+    digest.accept(Utils.toBytes(workflowAccession));
```

Comment:
These will have the same ids as the real actions, which...I'm not sure if that's a good idea or not. We should discuss that.